src/renderers/webgl/plugins/ShadowMapPlugin.js

```diff
@@ -318,7 +320,7 @@ THREE.ShadowMapPlugin = function ( _renderer, _lights, _webglObjects, _webglObje
 				webglObject = _webglObjectsImmediate[ j ];
 				object = webglObject.object;
 
-				if ( object.visible && object.castShadow ) {
+				if ( object.isVisible( shadowCamera ) && object.castShadow ) {
```

Comment:
Should I have the camera layers check here? This loop is about rendering immediate objects. This already checks if object casts shadows, so my guess is the layers check is needed, right?